lib/luacrypto_shash.c

```diff
@@ -0,0 +1,298 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for synchronous
+* message digest (hash) algorithms, including HMAC.
+*
+* This module provides a `new` function to create SHASH transform objects,
+* which can then be used for various hashing operations.
+*
+* @module crypto_shash
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/hash.h>
+#include <linux/err.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+static const lunatik_class_t luacrypto_shash_tfm_class;
+
+typedef struct {
+	struct shash_desc *sdesc; /* Points to kmalloc'ed shash_desc + space */
+} luacrypto_shash_tfm_t;
```

Comment:
I would call your type just `luacrypto_shash_t` for the sake of simplicity.. you have no name clashing with that and I think it's clear enough..